[2025-04-23 22:49:27,032: INFO: main: Welcome to cnnClassifier]
[2025-05-08 14:32:36,677: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 14:32:36,679: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 14:33:57,434: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 14:33:57,447: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 14:33:57,451: INFO: common: created directory at: artifacts]
[2025-05-08 14:33:57,454: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 14:33:59,880: INFO: 1170291011: artifacts/data_ingestion/data.zip download! with following info: 
Date: Thu, 08 May 2025 09:04:01 GMT
Content-Type: text/html; charset=utf-8
Vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame,Accept-Encoding, Accept, X-Requested-With
ETag: W/"96aeb85347b6be256f413ae0e457578d"
Cache-Control: max-age=0, private, must-revalidate
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 0
Referrer-Policy: no-referrer-when-downgrade
Content-Security-Policy: default-src 'none'; base-uri 'self'; child-src github.githubassets.com github.com/assets-cdn/worker/ github.com/assets/ gist.github.com/assets-cdn/worker/; connect-src 'self' uploads.github.com www.githubstatus.com collector.github.com raw.githubusercontent.com api.github.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com *.rel.tunnels.api.visualstudio.com wss://*.rel.tunnels.api.visualstudio.com objects-origin.githubusercontent.com copilot-proxy.githubusercontent.com proxy.individual.githubcopilot.com proxy.business.githubcopilot.com proxy.enterprise.githubcopilot.com *.actions.githubusercontent.com wss://*.actions.githubusercontent.com productionresultssa0.blob.core.windows.net/ productionresultssa1.blob.core.windows.net/ productionresultssa2.blob.core.windows.net/ productionresultssa3.blob.core.windows.net/ productionresultssa4.blob.core.windows.net/ productionresultssa5.blob.core.windows.net/ productionresultssa6.blob.core.windows.net/ productionresultssa7.blob.core.windows.net/ productionresultssa8.blob.core.windows.net/ productionresultssa9.blob.core.windows.net/ productionresultssa10.blob.core.windows.net/ productionresultssa11.blob.core.windows.net/ productionresultssa12.blob.core.windows.net/ productionresultssa13.blob.core.windows.net/ productionresultssa14.blob.core.windows.net/ productionresultssa15.blob.core.windows.net/ productionresultssa16.blob.core.windows.net/ productionresultssa17.blob.core.windows.net/ productionresultssa18.blob.core.windows.net/ productionresultssa19.blob.core.windows.net/ github-production-repository-image-32fea6.s3.amazonaws.com github-production-release-asset-2e65be.s3.amazonaws.com insights.github.com wss://alive.github.com api.githubcopilot.com api.individual.githubcopilot.com api.business.githubcopilot.com api.enterprise.githubcopilot.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com copilot-workspace.githubnext.com objects-origin.githubusercontent.com; frame-ancestors 'none'; frame-src viewscreen.githubusercontent.com notebooks.githubusercontent.com; img-src 'self' data: blob: github.githubassets.com media.githubusercontent.com camo.githubusercontent.com identicons.github.com avatars.githubusercontent.com private-avatars.githubusercontent.com github-cloud.s3.amazonaws.com objects.githubusercontent.com release-assets.githubusercontent.com secured-user-images.githubusercontent.com/ user-images.githubusercontent.com/ private-user-images.githubusercontent.com opengraph.githubassets.com copilotprodattachments.blob.core.windows.net/github-production-copilot-attachments/ github-production-user-asset-6210df.s3.amazonaws.com customer-stories-feed.github.com spotlights-feed.github.com objects-origin.githubusercontent.com *.githubusercontent.com; manifest-src 'self'; media-src github.com user-images.githubusercontent.com/ secured-user-images.githubusercontent.com/ private-user-images.githubusercontent.com github-production-user-asset-6210df.s3.amazonaws.com gist.github.com; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com; upgrade-insecure-requests; worker-src github.githubassets.com github.com/assets-cdn/worker/ github.com/assets/ gist.github.com/assets-cdn/worker/
Server: github.com
Accept-Ranges: bytes
Set-Cookie: _gh_sess=l2j7FDAJAgTd5LsH9iAwRMT5maehbkPFVA1XWx2AuawG2cn42AdecbIpvASRJDMeU9bfXbywvVdNQYvZ95lai2NzAT7fHk79BYO%2FoC5OLXVh0Cmw4pvCyv3361WyBnmzwVMGV270AWoDgj7%2BPX7RWkDHSKZLke%2FEjXHTQ05ub3a59jIeZY8zph3eehTMMll8I8iYRinXguUIrczWqRi16TUxkT6j%2BUOQrNEEJDEdBjCAg%2FuSW9z4da6I4PipODSV3uiRFLkIfojDHqN8kmGo0w%3D%3D--vKIGpfwQeXcHoDpI--C8a8mFATndmT9%2F09i0eqQA%3D%3D; Path=/; HttpOnly; Secure; SameSite=Lax
Set-Cookie: _octo=GH1.1.1515213014.1746695041; Path=/; Domain=github.com; Expires=Fri, 08 May 2026 09:04:01 GMT; Secure; SameSite=Lax
Set-Cookie: logged_in=no; Path=/; Domain=github.com; Expires=Fri, 08 May 2026 09:04:01 GMT; HttpOnly; Secure; SameSite=Lax
X-GitHub-Request-Id: 310B:3171DF:180BC9E:1D1543D:681C7380
connection: close
Transfer-Encoding: chunked

]
[2025-05-08 14:36:38,707: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 14:36:38,724: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 14:36:38,727: INFO: common: created directory at: artifacts]
[2025-05-08 14:36:38,727: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 14:36:38,731: INFO: 3130634179: File already exists of size: ~ 165 KB]
[2025-05-08 14:36:38,733: ERROR: 3130634179: The file artifacts/data_ingestion/data.zip is not a valid ZIP file.]
[2025-05-08 14:39:59,765: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 14:39:59,773: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 14:39:59,776: INFO: common: created directory at: artifacts]
[2025-05-08 14:39:59,777: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 14:39:59,779: INFO: 4119119724: Removing existing file: artifacts/data_ingestion/data.zip]
[2025-05-08 14:40:01,545: INFO: 4119119724: artifacts/data_ingestion/data.zip downloaded with the following info: 
Date: Thu, 08 May 2025 09:10:03 GMT
Content-Type: text/html; charset=utf-8
Vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame,Accept-Encoding, Accept, X-Requested-With
ETag: W/"da70ecaba09d89f6180de60e8696eeff"
Cache-Control: max-age=0, private, must-revalidate
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 0
Referrer-Policy: no-referrer-when-downgrade
Content-Security-Policy: default-src 'none'; base-uri 'self'; child-src github.githubassets.com github.com/assets-cdn/worker/ github.com/assets/ gist.github.com/assets-cdn/worker/; connect-src 'self' uploads.github.com www.githubstatus.com collector.github.com raw.githubusercontent.com api.github.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com *.rel.tunnels.api.visualstudio.com wss://*.rel.tunnels.api.visualstudio.com objects-origin.githubusercontent.com copilot-proxy.githubusercontent.com proxy.individual.githubcopilot.com proxy.business.githubcopilot.com proxy.enterprise.githubcopilot.com *.actions.githubusercontent.com wss://*.actions.githubusercontent.com productionresultssa0.blob.core.windows.net/ productionresultssa1.blob.core.windows.net/ productionresultssa2.blob.core.windows.net/ productionresultssa3.blob.core.windows.net/ productionresultssa4.blob.core.windows.net/ productionresultssa5.blob.core.windows.net/ productionresultssa6.blob.core.windows.net/ productionresultssa7.blob.core.windows.net/ productionresultssa8.blob.core.windows.net/ productionresultssa9.blob.core.windows.net/ productionresultssa10.blob.core.windows.net/ productionresultssa11.blob.core.windows.net/ productionresultssa12.blob.core.windows.net/ productionresultssa13.blob.core.windows.net/ productionresultssa14.blob.core.windows.net/ productionresultssa15.blob.core.windows.net/ productionresultssa16.blob.core.windows.net/ productionresultssa17.blob.core.windows.net/ productionresultssa18.blob.core.windows.net/ productionresultssa19.blob.core.windows.net/ github-production-repository-image-32fea6.s3.amazonaws.com github-production-release-asset-2e65be.s3.amazonaws.com insights.github.com wss://alive.github.com api.githubcopilot.com api.individual.githubcopilot.com api.business.githubcopilot.com api.enterprise.githubcopilot.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com copilot-workspace.githubnext.com objects-origin.githubusercontent.com; frame-ancestors 'none'; frame-src viewscreen.githubusercontent.com notebooks.githubusercontent.com; img-src 'self' data: blob: github.githubassets.com media.githubusercontent.com camo.githubusercontent.com identicons.github.com avatars.githubusercontent.com private-avatars.githubusercontent.com github-cloud.s3.amazonaws.com objects.githubusercontent.com release-assets.githubusercontent.com secured-user-images.githubusercontent.com/ user-images.githubusercontent.com/ private-user-images.githubusercontent.com opengraph.githubassets.com copilotprodattachments.blob.core.windows.net/github-production-copilot-attachments/ github-production-user-asset-6210df.s3.amazonaws.com customer-stories-feed.github.com spotlights-feed.github.com objects-origin.githubusercontent.com *.githubusercontent.com; manifest-src 'self'; media-src github.com user-images.githubusercontent.com/ secured-user-images.githubusercontent.com/ private-user-images.githubusercontent.com github-production-user-asset-6210df.s3.amazonaws.com gist.github.com; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com; upgrade-insecure-requests; worker-src github.githubassets.com github.com/assets-cdn/worker/ github.com/assets/ gist.github.com/assets-cdn/worker/
Server: github.com
Accept-Ranges: bytes
Set-Cookie: _gh_sess=TFmj8tAfd1zPFn9nAOVidtZizbtoTAj1Z4uvcTRs6RJBI8SmyM%2Be7Er3yIN%2Fi%2FmL7Sr493%2FHAxX47kQqLNAeeYtcuQQuH8uTBm7ex29NxzX34lTWb6a5zCBV5PxTco9g4BTB%2BhjoFu%2FsJ%2FchU1TrWtPh%2F7pisYstkR259je6hOcBBDSs2SngyPziawZdMmch4hKWnMdKbKd7MlmYHMJvsWhLt2F5JMoGw5GaoJDZ4uw10fD2h1i9vHpHZxC4pOYe8MVetNOALYPNC%2BK%2F7Y5%2FKg%3D%3D--DLtIloWK3l8C9q2v--WRgcZb7yd6MfD0fqUvKk1A%3D%3D; Path=/; HttpOnly; Secure; SameSite=Lax
Set-Cookie: _octo=GH1.1.35474465.1746695403; Path=/; Domain=github.com; Expires=Fri, 08 May 2026 09:10:03 GMT; Secure; SameSite=Lax
Set-Cookie: logged_in=no; Path=/; Domain=github.com; Expires=Fri, 08 May 2026 09:10:03 GMT; HttpOnly; Secure; SameSite=Lax
X-GitHub-Request-Id: D68A:1D6E92:585409:6805AD:681C74EB
connection: close
Transfer-Encoding: chunked

]
[2025-05-08 14:40:01,551: INFO: 4119119724: Downloaded file size: 169370 bytes]
[2025-05-08 14:40:01,551: ERROR: 4119119724: The file artifacts/data_ingestion/data.zip is not a valid ZIP file.]
[2025-05-08 16:11:47,011: INFO: main: Welcome to cnnClassifier]
[2025-05-08 16:11:47,012: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 16:11:47,013: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 16:11:47,015: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 16:11:47,016: INFO: common: created directory at: artifacts]
[2025-05-08 16:11:47,017: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 16:11:47,018: INFO: data_ingestion: File already exists of size: ~ 165 KB]
[2025-05-08 16:11:47,021: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 11, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1268, in __init__
    self._RealGetContents()
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1335, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-05-08 16:38:48,136: INFO: main: Welcome to cnnClassifier]
[2025-05-08 16:38:48,136: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 16:38:48,136: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 16:38:48,136: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 16:38:48,136: INFO: common: created directory at: artifacts]
[2025-05-08 16:38:48,146: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 16:38:48,147: INFO: data_ingestion: File already exists of size: ~ 165 KB]
[2025-05-08 16:38:48,150: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 12, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1268, in __init__
    self._RealGetContents()
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1335, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-05-08 16:45:12,926: INFO: main: Welcome to cnnClassifier]
[2025-05-08 16:45:12,926: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 16:45:12,932: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 16:45:12,932: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 16:45:12,932: INFO: common: created directory at: artifacts]
[2025-05-08 16:45:12,932: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 16:45:12,932: INFO: data_ingestion: File already exists of size: ~ 165 KB]
[2025-05-08 16:45:12,938: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 12, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1268, in __init__
    self._RealGetContents()
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1335, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-05-08 16:47:50,763: INFO: main: Welcome to cnnClassifier]
[2025-05-08 16:47:50,763: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 16:47:50,766: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 16:47:50,766: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 16:47:50,766: INFO: common: created directory at: artifacts]
[2025-05-08 16:47:50,766: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 16:47:50,766: INFO: data_ingestion: File already exists of size: ~ 165 KB]
[2025-05-08 16:47:50,772: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 12, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1268, in __init__
    self._RealGetContents()
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1335, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-05-08 16:52:39,021: INFO: main: Welcome to cnnClassifier]
[2025-05-08 16:52:39,021: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 16:52:39,023: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 16:52:39,024: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 16:52:39,025: INFO: common: created directory at: artifacts]
[2025-05-08 16:52:39,026: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 16:52:39,026: INFO: data_ingestion: File already exists of size: ~ 165 KB]
[2025-05-08 16:52:39,028: ERROR: main: File is not a zip file]
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 12, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.extract_zip_file()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\components\data_ingestion.py", line 37, in extract_zip_file
    with zipfile.ZipFile(self.config.local_data_file, 'r') as zip_ref:
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1268, in __init__
    self._RealGetContents()
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\zipfile.py", line 1335, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-05-08 16:54:53,130: INFO: main: Welcome to cnnClassifier]
[2025-05-08 16:54:53,130: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 16:54:53,132: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 16:54:53,134: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 16:54:53,135: INFO: common: created directory at: artifacts]
[2025-05-08 16:54:53,135: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 16:54:53,135: INFO: data_ingestion: File already exists of size: ~ 11345 KB]
[2025-05-08 16:54:53,345: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 18:33:56,618: INFO: main: Welcome to cnnClassifier]
[2025-05-08 18:33:56,618: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 18:33:56,618: ERROR: main: [Errno 2] No such file or directory: 'C:\\Users\\abhis\\OneDrive\\Desktop\\Projects\\end-to-end-cdc\\chicken-diseases-classification\\src\\cnnClassifier\\config\\config.yaml']
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 12, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\config\configuration.py", line 18, in __init__
    self.config = read_yaml(config_filepath)
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\site-packages\ensure\main.py", line 849, in __call__
    return_val = self.f(*args, **kwargs)
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\utils\common.py", line 37, in read_yaml
    raise e
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\utils\common.py", line 30, in read_yaml
    with open(path_to_yaml) as yaml_file:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\abhis\\OneDrive\\Desktop\\Projects\\end-to-end-cdc\\chicken-diseases-classification\\src\\cnnClassifier\\config\\config.yaml'
[2025-05-08 19:08:13,713: INFO: main: Welcome to cnnClassifier]
[2025-05-08 19:08:13,713: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 19:08:13,713: INFO: common: yaml file: C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\config\config.yaml loaded successfully]
[2025-05-08 19:08:13,713: ERROR: main: [Errno 2] No such file or directory: 'C:\\Users\\abhis\\OneDrive\\Desktop\\Projects\\end-to-end-cdc\\chicken-diseases-classification\\src\\cnnClassifier\\params.yaml']
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 12, in <module>
    data_ingestion.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\config\configuration.py", line 19, in __init__
    self.params = read_yaml(params_filepath)
  File "C:\Users\abhis\anaconda3\envs\cdc\lib\site-packages\ensure\main.py", line 849, in __call__
    return_val = self.f(*args, **kwargs)
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\utils\common.py", line 37, in read_yaml
    raise e
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\utils\common.py", line 30, in read_yaml
    with open(path_to_yaml) as yaml_file:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\abhis\\OneDrive\\Desktop\\Projects\\end-to-end-cdc\\chicken-diseases-classification\\src\\cnnClassifier\\params.yaml'
[2025-05-08 19:46:51,192: INFO: main: Welcome to cnnClassifier]
[2025-05-08 19:46:51,192: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 19:46:51,192: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 19:46:51,192: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 19:46:51,198: INFO: common: created directory at: artifacts]
[2025-05-08 19:46:51,198: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 19:46:59,589: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 11616915
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "adf745abc03891fe493c3be264ec012691fe3fa21d861f35a27edbe6d86a76b1"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: E303:B24B:18764:30E80:681CB9B8
Accept-Ranges: bytes
Date: Thu, 08 May 2025 14:16:55 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10238-MAA
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1746713815.289583,VS0,VE292
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 8245266b55ae3c8e9ebf8de0188b59ff39060f5d
Expires: Thu, 08 May 2025 14:21:55 GMT
Source-Age: 0

]
[2025-05-08 19:46:59,941: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 21:27:27,310: INFO: main: Welcome to cnnClassifier]
[2025-05-08 21:27:27,310: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 21:27:27,320: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 21:27:27,320: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 21:27:27,320: INFO: common: created directory at: artifacts]
[2025-05-08 21:27:27,324: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 21:28:07,541: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 11616915
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "adf745abc03891fe493c3be264ec012691fe3fa21d861f35a27edbe6d86a76b1"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 5E73:2AA34C:3D53B:7EB00:681CD46B
Accept-Ranges: bytes
Date: Thu, 08 May 2025 15:57:32 GMT
Via: 1.1 varnish
X-Served-By: cache-maa10230-MAA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1746719852.048394,VS0,VE891
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 1dd068f7a37e145cae31979a7ff7586149724af1
Expires: Thu, 08 May 2025 16:02:32 GMT
Source-Age: 1

]
[2025-05-08 21:28:07,914: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 21:28:07,914: INFO: main: *******************]
[2025-05-08 21:28:07,914: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2025-05-08 21:28:07,914: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 21:28:07,914: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 21:28:07,920: INFO: common: created directory at: artifacts]
[2025-05-08 21:28:07,920: INFO: common: created directory at: artifacts/prepare_base_model]
[2025-05-08 21:28:08,530: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2025-05-08 21:28:08,815: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2025-05-08 21:30:36,974: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 21:30:36,974: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 21:30:36,979: INFO: common: created directory at: artifacts]
[2025-05-08 21:30:36,982: INFO: common: created directory at: artifacts/prepare_base_model]
[2025-05-08 21:30:37,674: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2025-05-08 21:48:40,304: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 21:48:40,311: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 21:48:40,314: INFO: common: created directory at: artifacts]
[2025-05-08 21:48:40,318: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2025-05-08 21:48:40,321: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2025-05-08 21:59:32,431: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 21:59:32,436: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 21:59:32,439: INFO: common: created directory at: artifacts]
[2025-05-08 21:59:32,441: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2025-05-08 21:59:32,444: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2025-05-08 21:59:32,446: INFO: common: created directory at: artifacts\training]
[2025-05-08 22:04:42,139: INFO: main: Welcome to cnnClassifier]
[2025-05-08 22:04:42,140: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 22:04:42,142: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:04:42,143: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:04:42,144: INFO: common: created directory at: artifacts]
[2025-05-08 22:04:42,144: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 22:04:42,145: INFO: data_ingestion: File already exists of size: ~ 11345 KB]
[2025-05-08 22:04:42,395: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 22:04:42,395: INFO: main: *******************]
[2025-05-08 22:04:42,396: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2025-05-08 22:04:42,397: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:04:42,399: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:04:42,400: INFO: common: created directory at: artifacts]
[2025-05-08 22:04:42,400: INFO: common: created directory at: artifacts/prepare_base_model]
[2025-05-08 22:04:42,809: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2025-05-08 22:04:43,039: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2025-05-08 22:04:43,039: INFO: main: *******************]
[2025-05-08 22:04:43,040: INFO: main: >>>>>> stage Training started <<<<<<]
[2025-05-08 22:04:43,043: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:04:43,045: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:04:43,046: INFO: common: created directory at: artifacts]
[2025-05-08 22:04:43,047: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2025-05-08 22:04:43,049: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2025-05-08 22:04:43,051: INFO: common: created directory at: artifacts\training]
[2025-05-08 22:05:24,333: INFO: main: >>>>>> stage Training completed <<<<<<

x==========x]
[2025-05-08 22:34:41,757: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:34:41,763: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:34:41,765: INFO: common: created directory at: artifacts]
[2025-05-08 22:35:38,490: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:35:38,494: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:35:38,495: INFO: common: created directory at: artifacts]
[2025-05-08 22:55:53,048: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:55:53,059: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:55:53,062: INFO: common: created directory at: artifacts]
[2025-05-08 22:57:53,102: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 22:57:53,106: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 22:57:53,109: INFO: common: created directory at: artifacts]
[2025-05-08 23:01:34,271: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:01:34,276: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:01:34,279: INFO: common: created directory at: artifacts]
[2025-05-08 23:03:02,064: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:03:02,068: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:03:02,070: INFO: common: created directory at: artifacts]
[2025-05-08 23:03:44,038: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:03:44,041: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:03:44,043: INFO: common: created directory at: artifacts]
[2025-05-08 23:06:05,037: INFO: main: Welcome to cnnClassifier]
[2025-05-08 23:06:05,037: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 23:06:05,045: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:06:05,045: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:06:05,045: INFO: common: created directory at: artifacts]
[2025-05-08 23:06:05,048: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 23:06:05,048: INFO: data_ingestion: File already exists of size: ~ 11345 KB]
[2025-05-08 23:06:05,483: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 23:06:05,483: INFO: main: *******************]
[2025-05-08 23:06:05,487: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2025-05-08 23:06:05,487: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:06:05,487: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:06:05,487: INFO: common: created directory at: artifacts]
[2025-05-08 23:06:05,493: INFO: common: created directory at: artifacts/prepare_base_model]
[2025-05-08 23:06:05,937: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2025-05-08 23:06:06,248: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2025-05-08 23:06:06,248: INFO: main: *******************]
[2025-05-08 23:06:06,248: INFO: main: >>>>>> stage Training started <<<<<<]
[2025-05-08 23:06:06,253: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:06:06,253: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:06:06,253: INFO: common: created directory at: artifacts]
[2025-05-08 23:06:06,253: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2025-05-08 23:06:06,258: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2025-05-08 23:06:06,260: INFO: common: created directory at: artifacts\training]
[2025-05-08 23:06:53,240: INFO: main: >>>>>> stage Training completed <<<<<<

x==========x]
[2025-05-08 23:06:53,240: INFO: main: *******************]
[2025-05-08 23:06:53,240: INFO: main: >>>>>> stage Evaluation stage started <<<<<<]
[2025-05-08 23:06:53,247: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:06:53,255: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:06:53,257: INFO: common: created directory at: artifacts]
[2025-05-08 23:06:53,257: ERROR: main: __init__() missing 1 required positional argument: 'mlflow_uri']
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 51, in <module>
    model_evalution.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_04_evaluation.py", line 17, in main
    eval_config = config.get_evaluation_config()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\config\configuration.py", line 99, in get_evaluation_config
    eval_config = EvaluationConfig(
TypeError: __init__() missing 1 required positional argument: 'mlflow_uri'
[2025-05-08 23:09:23,138: INFO: main: Welcome to cnnClassifier]
[2025-05-08 23:09:23,138: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 23:09:23,138: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:09:23,148: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:09:23,148: INFO: common: created directory at: artifacts]
[2025-05-08 23:09:23,148: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 23:09:23,150: INFO: data_ingestion: File already exists of size: ~ 11345 KB]
[2025-05-08 23:09:23,419: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 23:09:23,419: INFO: main: *******************]
[2025-05-08 23:09:23,419: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2025-05-08 23:09:23,422: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:09:23,423: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:09:23,423: INFO: common: created directory at: artifacts]
[2025-05-08 23:09:23,425: INFO: common: created directory at: artifacts/prepare_base_model]
[2025-05-08 23:09:23,882: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2025-05-08 23:09:24,112: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2025-05-08 23:09:24,112: INFO: main: *******************]
[2025-05-08 23:09:24,112: INFO: main: >>>>>> stage Training started <<<<<<]
[2025-05-08 23:09:24,118: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:09:24,120: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:09:24,120: INFO: common: created directory at: artifacts]
[2025-05-08 23:09:24,120: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2025-05-08 23:09:24,120: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2025-05-08 23:09:24,120: INFO: common: created directory at: artifacts\training]
[2025-05-08 23:10:08,229: INFO: main: >>>>>> stage Training completed <<<<<<

x==========x]
[2025-05-08 23:10:08,229: INFO: main: *******************]
[2025-05-08 23:10:08,229: INFO: main: >>>>>> stage Evaluation stage started <<<<<<]
[2025-05-08 23:10:08,235: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:10:08,237: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:10:08,237: INFO: common: created directory at: artifacts]
[2025-05-08 23:10:24,318: INFO: common: json file saved at: scores.json]
[2025-05-08 23:10:24,319: ERROR: main: 'EvaluationConfig' object has no attribute 'mlflow_uri']
Traceback (most recent call last):
  File "C:\Users\abhis\OneDrive\Desktop\Projects\end-to-end-cdc\chicken-diseases-classification\main.py", line 51, in <module>
    model_evalution.main()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\pipeline\stage_04_evaluation.py", line 21, in main
    evaluation.log_into_mlflow()
  File "c:\users\abhis\onedrive\desktop\projects\end-to-end-cdc\chicken-diseases-classification\src\cnnClassifier\components\evaluation.py", line 58, in log_into_mlflow
    mlflow.set_registry_uri(self.config.mlflow_uri)
AttributeError: 'EvaluationConfig' object has no attribute 'mlflow_uri'
[2025-05-08 23:13:11,177: INFO: main: Welcome to cnnClassifier]
[2025-05-08 23:13:11,177: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2025-05-08 23:13:11,177: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:13:11,177: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:13:11,185: INFO: common: created directory at: artifacts]
[2025-05-08 23:13:11,185: INFO: common: created directory at: artifacts/data_ingestion]
[2025-05-08 23:13:11,185: INFO: data_ingestion: File already exists of size: ~ 11345 KB]
[2025-05-08 23:13:11,407: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2025-05-08 23:13:11,407: INFO: main: *******************]
[2025-05-08 23:13:11,407: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2025-05-08 23:13:11,407: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:13:11,407: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:13:11,407: INFO: common: created directory at: artifacts]
[2025-05-08 23:13:11,407: INFO: common: created directory at: artifacts/prepare_base_model]
[2025-05-08 23:13:11,852: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2025-05-08 23:13:12,208: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2025-05-08 23:13:12,208: INFO: main: *******************]
[2025-05-08 23:13:12,208: INFO: main: >>>>>> stage Training started <<<<<<]
[2025-05-08 23:13:12,217: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:13:12,217: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:13:12,217: INFO: common: created directory at: artifacts]
[2025-05-08 23:13:12,221: INFO: common: created directory at: artifacts\prepare_callbacks\checkpoint_dir]
[2025-05-08 23:13:12,221: INFO: common: created directory at: artifacts\prepare_callbacks\tensorboard_log_dir]
[2025-05-08 23:13:12,221: INFO: common: created directory at: artifacts\training]
[2025-05-08 23:13:54,064: INFO: main: >>>>>> stage Training completed <<<<<<

x==========x]
[2025-05-08 23:13:54,064: INFO: main: *******************]
[2025-05-08 23:13:54,064: INFO: main: >>>>>> stage Evaluation stage started <<<<<<]
[2025-05-08 23:13:54,067: INFO: common: yaml file: config\config.yaml loaded successfully]
[2025-05-08 23:13:54,067: INFO: common: yaml file: params.yaml loaded successfully]
[2025-05-08 23:13:54,067: INFO: common: created directory at: artifacts]
[2025-05-08 23:14:08,891: INFO: common: json file saved at: scores.json]
[2025-05-08 23:14:08,891: INFO: main: >>>>>> stage Evaluation stage completed <<<<<<

x==========x]
